diff --git a/CMakeLists.txt b/CMakeLists.txt
index 131cc96e..d34531db 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,9 +1,6 @@
 cmake_minimum_required(VERSION 3.16)   
 
 project("UnityHubNative")
-set(CMAKE_INSTALL_PREFIX ${CMAKE_CURRENT_BINARY_DIR})
-set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/$<CONFIGURATION>)
-set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/$<CONFIGURATION>)
 
 option(BUILD_SHARED_LIBS OFF "Build static")
 
@@ -94,20 +91,6 @@ if(APPLE)
 		" 
 		COMPONENT Runtime
 	)
-elseif(LINUX)
-INSTALL(CODE 
-	"include(${CMAKE_CURRENT_SOURCE_DIR}/appimage.cmake)
-	make_appimage(
-		EXE \"${CMAKE_INSTALL_PREFIX}/$<CONFIGURATION>/${PROJECT_NAME}\"
-		NAME \"${PROJECT_NAME}\"
-		ICON \"${CMAKE_CURRENT_SOURCE_DIR}/source/wxlin.xpm\"
-		DIR_ICON \"${CMAKE_CURRENT_SOURCE_DIR}/source/wxlin.xpm\"
-		OUTPUT_NAME \"${CMAKE_INSTALL_PREFIX}/$<CONFIGURATION>/${PROJECT_NAME}.AppImage\"
-		ASSETS \"\"
-	)
-	" 
-	COMPONENT Runtime
-	)
 endif()
 # windows app
 set_property(TARGET ${PROJECT_NAME} PROPERTY VS_DPI_AWARE "PerMonitor")
@@ -115,9 +98,7 @@ set_property(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY VS_STARTUP_PROJECT "
 
  # Strip binary for release builds
  if (CMAKE_BUILD_TYPE STREQUAL Release)
- if(LINUX)
-	 add_custom_command(TARGET ${PROJECT_NAME} POST_BUILD COMMAND ${CMAKE_STRIP} ${CMAKE_BUILD_TYPE}/${PROJECT_NAME})
- elseif(APPLE)
+ if(APPLE)
 	 add_custom_command(TARGET ${PROJECT_NAME} POST_BUILD COMMAND ${CMAKE_STRIP} ${CMAKE_BUILD_TYPE}/${PROJECT_NAME}.app/Contents/MacOS/${PROJECT_NAME})
  endif()
 endif ()
